R/layer_add_forecast_date.R

```diff
@@ -2,18 +2,20 @@
 #'
 #' @param frosting a `frosting` postprocessor
 #' @param forecast_date The forecast date to add as a column to the `epi_df`.
```

Comment:
Oops. Sorry, I missed this. To my mind the `forecast_date` is the date on which the forecast is made. So, by default, it should be `max(time_value)` from the training data. The `target_date` should be "the date the forecast is for". So that one should be `max(time_value) + ahead` by default. 

It looks like they're both the same currently right?

Reply:
Almost… Current defaults: `forecast date = max_time_value + ahead` (where `max_time_value` is for test data), `target_date = time_value + ahead` (based on simple.forecasts.Rmd). But I will change to what you specified. A couple qs for that...

For `forecast_date`, to get the max time value from the training data, is that using mold from components? Then, I check if the `forecast_date < as_of_date` of the test data and throw a warning there (that says "forecast_date is less than the most recent update date of the data.”, yes?

For `target_date`, by default, that is the max time value in the in the **test** data + ahead or no?

Reply:
The difficulty here is that leading/lagging increments the `time_value`. So this could all be a bit dangerous. In an ideal setting:
1. `target_date` would be the max time value in the test data + ahead, as you said.
2. `forecast_date` would be the max time value in the test data. (So seemingly, the date of the most recent data available to you). Acting like we have data up to (and including) today, and we produce a forecast today, then this should work.
However, lots of weird crud can happen that would screw these up. If data isn't available for today but only for yesterday, then that would throw things off. If we accidentally lead the test data, then it'll produce "future" time_values.